llvm/lib/IR/BasicBlock.cpp

```diff
@@ -1009,7 +1009,7 @@ void BasicBlock::spliceDebugInfoImpl(BasicBlock::iterator Dest, BasicBlock *Src,
     // generate the iterator with begin() / getFirstInsertionPt(), it means
     // any trailing debug-info at the end of the block would "normally" have
     // been pushed in front of "First". Move it there now.
-    DbgMarker *FirstMarker = getMarker(First);
+    DbgMarker *FirstMarker = createMarker(First);
```

Comment:
Can you drop this into the `if (TrailingDbgRecords)` body? We don't want to create a marker if we're not going to use it when `TrailingDbgRecords` is false.

Reply:
done.